bundles/admin/admin-announcements/view/AnnouncementsListHandler.js

```diff
@@ -0,0 +1,156 @@
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    // Update all the announcements f.ex. when saved. Set active key as empty so all panels get closed.
+    updateAnnouncement (data) {
+        this.announcementsHelper.updateAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.updateFailed'));
+                return false;
+            } else {
+                Messaging.success(getMessage('messages.updateSuccess'));
+                removeFromLocalStorageArray(ANNOUNCEMENTS_LOCALSTORAGE, data.id);
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    deleteAnnouncement (index, id) {
+        this.id = { id };
+        this.announcementsHelper.deleteAnnouncement(this.id, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.deleteFailed'));
+            } else {
+                Messaging.success(getMessage('messages.deleteSuccess'));
+                const newList = [...this.state.announcements];
+                newList.splice(index, 1);
+                // Update accordion with announcements and keep all closed
+                this.updateState({
+                    announcements: newList,
+                    activeKey: [],
+                    updated: false
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+    // Cancel creating a new announcement or editing one. Close all panels.
+    cancel (index, id) {
+        if (id !== undefined) {
+            this.updateState({
+                activeKey: []
+            });
+        } else {
+            const newList = [...this.state.announcements];
+            newList.splice(index, 1);
+            this.updateState({
+                announcements: newList,
+                activeKey: []
+            });
+        }
+    }
+
+    addForm () {
+        const newList = [...this.state.announcements];
+        newList.push({
+            title: this.newTitle
+        });
+        this.updateState({
+            announcements: newList,
+            title: this.newTitle
+        });
+    }
+
+    updateTitle (value) {
+        this.updateState({
+            title: value
+        });
+    }
+    toggleActive () {
+        this.updateState({
+            active: !this.state.active
+        });
+    }
+
+    updateActiveKey (key) {
+        this.updateState({
+            activeKey: key
+        });
+    }
+}
+
+export const AnnouncementsListHandler = controllerMixin(ViewHandler, [
+    'pushAnnouncements', 'getAdminAnnouncements', 'addForm', 'deleteAnnouncement', 'updateTitle', 'toggleActive', 'saveAnnouncement', 'updateAnnouncement', 'cancel', 'updateActiveKey'
```

Comment:
Can you check this list to only include existing functions that are not getters.